pkg/vault/contracts/VaultExplorer.sol

```diff
@@ -0,0 +1,333 @@
+    function computeDynamicSwapFee(
+        address pool,
+        IBasePool.PoolSwapParams memory swapParams
+    ) external view returns (bool success, uint256 dynamicSwapFee) {
+        return _vault.computeDynamicSwapFee(pool, swapParams);
+    }
+
+    /*******************************************************************************
+                                    Recovery Mode
+    *******************************************************************************/
+
+    /// @inheritdoc IVaultExplorer
+    function isPoolInRecoveryMode(address pool) external view returns (bool) {
+        return _vault.isPoolInRecoveryMode(pool);
+    }
+
+    /*******************************************************************************
+                                    Queries
+    *******************************************************************************/
+
+    /// @inheritdoc IVaultExplorer
+    function isQueryDisabled() external view returns (bool) {
+        return _vault.isQueryDisabled();
+    }
+
+    /***************************************************************************
+                              Vault Admin Functions
+    ***************************************************************************/
+
+    /// @inheritdoc IVaultExplorer
+    function getPauseWindowEndTime() external view returns (uint32) {
+        return _vault.getPauseWindowEndTime();
+    }
+
+    /// @inheritdoc IVaultExplorer
+    function getBufferPeriodDuration() external view returns (uint32) {
+        return _vault.getBufferPeriodDuration();
+    }
+
+    /// @inheritdoc IVaultExplorer
+    function getBufferPeriodEndTime() external view returns (uint32) {
+        return _vault.getBufferPeriodEndTime();
+    }
+
+    /// @inheritdoc IVaultExplorer
+    function getMinimumPoolTokens() external view returns (uint256) {
+        return _vault.getMinimumPoolTokens();
+    }
+
+    /// @inheritdoc IVaultExplorer
+    function getMaximumPoolTokens() external view returns (uint256) {
+        return _vault.getMaximumPoolTokens();
+    }
+
+    /*******************************************************************************
+                                    Vault Pausing
+    *******************************************************************************/
+
+    /// @inheritdoc IVaultExplorer
+    function isVaultPaused() external view returns (bool) {
+        return _vault.isVaultPaused();
+    }
+
+    /// @inheritdoc IVaultExplorer
+    function getVaultPausedState() external view returns (bool, uint32, uint32) {
+        return _vault.getVaultPausedState();
+    }
+
+    /*******************************************************************************
+                                        Fees
+    *******************************************************************************/
+
+    /// @inheritdoc IVaultExplorer
+    function collectAggregateFees(address pool) external {
+        return _vault.collectAggregateFees(pool);
+    }
```

Comment:
You can also `registerPool`. That's why I was questioning the name. Probably people wouldn't be registering pools outside factories, and presumably collecting fees would be done programmatically somehow as well. "Explorer" does imply read-only. It could be "Facade" or "Interface" or "API" - but API is confusable, and interface is also confusable with the technical meaning.

I can remove those for now; we can always add them in later, since the tests, etc. are in source control.

Reply:
Per discussion, we'll take out registerPool, which doesn't make a lot of sense (should be done through a factory - how did they deploy the pool they're registering?), but leave the fee collection, which doesn't hurt anything.